src/h5web/toolbar/controls/DomainSlider/DomainSlider.tsx

```diff
@@ -38,38 +29,55 @@ function DomainSlider(props: Props): ReactElement {
   const { dataDomain, customDomain, scaleType, disabled } = props;
   const { onCustomDomainChange } = props;
 
-  const [visDomain, errors] = useVisDomain(dataDomain, customDomain, scaleType);
-  const [sliderDomain, setSliderDomain] = useState(visDomain);
+  const visDomain = useVisDomain(customDomain, dataDomain);
+  const [safeDomain, errors] = useSafeDomain(visDomain, dataDomain, scaleType);
 
+  const [sliderDomain, setSliderDomain] = useState(visDomain);
```

Comment:
The slider domain state now holds the "unsafe" vis domain. This allows the tooltip to show the custom domain even when it's invalid, while also showing the slider's domain while a thumb is being moved.

The `safeDomain` is passed to the `ScaledSlider` for computing the extended domain. `ScaledSlider` then recomputes a safe domain from its `value` on every render so the thumbs don't end up in weird places.